Yank/utils.py

```diff
@@ -845,6 +855,170 @@ def camelcase_to_underscore(camelcase_str):
     underscore_str = re.sub(r'([A-Z])', '_\g<1>', camelcase_str)
     return underscore_str.lower()
 
+def quantity_from_string(quantity_str):
```

Comment:
Should I write explicit tests for this parser and/or make a simpler parser?

This is a powerful converter that can take arbitrary strings and covert them to `units.Quantity`s based on a number of valid operators. Its automatically (although implicitly) tested by a number of the other tests. It can parse more complex strings than any of the other tests or use cases call for right now though.